examples/settings/settings_main.c

```
@@ -0,0 +1,276 @@
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * settings_main
+ ****************************************************************************/
+
+int main(int argc, FAR char *argv[])
+{
+  int                  ret;
+  int                  fd;
+  enum settings_type_e stype;
+  char                 path[50];
+  char                 *teststr = "";
+  int                  testval = 0x5aa5;
+  char                 readstr[CONFIG_SETTINGS_VALUE_SIZE];
+  enum storage_type_e  storage_type = STORAGE_BINARY;
+  bool                 flag_present = false;
+  struct stat          sbuf;
+
+  ASSERT(0);
+  if ((argc < 1) || *argv[1] == 0 || *(argv[1] + 1) == 0)
+    {
+      goto print_help;
+    }
+
+  if (*argv[1] == '-')
+    {
+      flag_present = true;
+      if (*(argv[1] + 1) == 'b')
+        {
+          storage_type = STORAGE_BINARY;
+        }
+      else if (*(argv[1] + 1) == 't')
+        {
+          storage_type = STORAGE_TEXT;
+        }
+      else
+        {
+          goto print_help;
+        }
+    }
+
+  if (flag_present && argc < 2)
+    {
+      goto print_help;
+    }
+
+#ifdef CONFIG_EXAMPLES_SETTINGS_USE_TMPFS
+#  ifndef CONFIG_FS_TMPFS
+#    error TMPFS must be enabled
+#  endif
+#  ifndef CONFIG_LIBC_TMPDIR
+#    error LIBC_TMPDIR must be defined
+#  endif
+  if (stat(CONFIG_LIBC_TMPDIR, &sbuf))
+    {
+      ret = nx_mount(NULL, CONFIG_LIBC_TMPDIR, "tmpfs", 0, NULL);
+      if (ret < 0)
+        {
+          printf("ERROR: Failed to mount tmpfs at %s: %d\n",
+                  CONFIG_LIBC_TMPDIR, ret);
+          goto end;
```

This will eventually `return OK;`. Maybe it's better to return an error in all these cases?

---

So, it is the "return OK" at the end of the app that is wrong. That should be `return ret`